FryProxy/src/Writers/HttpMessageWriter.cs

```diff
@@ -60,12 +60,13 @@ public void Write(HttpMessageHeader header, Stream body = null, Nullable<Int64>
             if (body == null)
             {
                 return;
-            } 
-            
-            WriteBody(header, body, bodyLength.GetValueOrDefault(0));
-            
-            writer.WriteLine();
-            writer.Flush();
+            }
+
+            if (!WriteBody(header, body, bodyLength.GetValueOrDefault(0)))

```

Comment:
Strange condition. I expected that we should add closing empty line and flush stream when the body was written and do not do it otherwise.